Update the arm-* and aarch64-* platform docs.

thejpster opened the pull request:
This PR updates some of the arm*-unknown-none target docs, and adds some missing target pages.

## aarch64-none-elf and aarch64-none-elf-softfloat

The Rust Embedded Devices Working Group's Arm Team is added as a maintainer, and a target page is added. Links are added to the EDWG's support crates for this target.

## armv7a-none-eabi and armv7a-none-eabihf

The Rust Embedded Devices Working Group's Arm Team is added as a maintainer, and a target page is added. Links are added to the EDWG's support crates for this target.

## armv7r-none-eabi and armv7r-none-eabihf

The Rust Embedded Devices Working Group's Arm Team is added as a maintainer, and the target page is split from the Big Endian versions. Links are added to the EDWG's support crates for this target.

## armebv7r-none-eabi and armveb7r-none-eabihf

The target page is split from the Little Endian versions. No change in maintainers.

I have agreement to add EDWG/T-Arm as maintainers, which was voted upon in [their repo](https://github.com/rust-embedded/wg/issues/851).

thejpster committed:
Update the arm-* and aarch64-* platform docs.

The Rust Embedded Devices Working Group (wg-embedded) Arm Team (t-arm)
agreed to listed as maintainers of:

* aarch64-unknown-none
* aarch64-unknown-none-softfloat
* armv7a-none-eabi
* armv7r-none-eabi
* armv7r-none-eabihf

The aarch64-unknown-none* target didn't have a page so I added it.

wg-embedded t-arm did not want to take over:

* armebv7r-none-eabi
* armebv7r-none-eabihf

So I gave them their own target page. The current maintainer remains.

rust-log-analyzer commented:
The job **`aarch64-gnu-llvm-19-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/17623359605/job/50074112515) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/50074112515) [(plain)](https://github.com/rust-lang/rust/commit/87a4edb1d71ee583eb03fc5ea4daa5b2c35241d0/checks/50074112515/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
tidy check
Checking tidy rustdoc_json...
No error code explanation was removed!
tidy: Skipping binary file check, read-only filesystem
##[error]tidy error: /checkout/src/doc/rustc/src/platform-support/armv7a-none-eabi.md:29: trailing whitespace
tidy error: /checkout/src/doc/rustc/src/platform-support/armv7r-none-eabi.md: too many trailing newlines (2)
some tidy checks failed
Command `/checkout/obj/build/aarch64-unknown-linux-gnu/stage1-tools-bin/rust-tidy /checkout /checkout/obj/build/aarch64-unknown-linux-gnu/stage0/bin/cargo /checkout/obj/build 4 npm` failed with exit code 1
Created at: src/bootstrap/src/core/build_steps/tool.rs:1550:23
Executed at: src/bootstrap/src/core/build_steps/test.rs:1281:29

Command has failed. Rerun with -v to see more details.
Bootstrap failed while executing `--stage 2 test --skip tests --skip coverage-map --skip coverage-run --skip library --skip tidyselftest`
Build completed unsuccessfully in 0:00:47
  local time: Wed Sep 10 18:36:48 UTC 2025
  network time: Wed, 10 Sep 2025 18:36:49 GMT
##[error]Process completed with exit code 1.
```

</details>

fbstj commented on src/doc/rustc/src/platform-support/armv7a-none-eabi.md:
```diff
@@ -0,0 +1,51 @@
+
+All Armv7-A processors include an FPU (a VFPv3 or a VFPv4). The difference
+between the `-eabi` and `-eabihf` targets is whether the FPU is used for
+passing function arguments. You may prefer the `-eabi` soft-float target when
```
nit: unfinished sentence

thejpster replied:
Oops. Fixed.

thejpster committed:
Pick up changes from robamu that I missed.

From https://github.com/thejpster/rust/pull/1.

rust-log-analyzer commented:
The job **`aarch64-gnu-llvm-19-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/17624754848/job/50078765985) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/50078765985) [(plain)](https://github.com/rust-lang/rust/commit/f1abb70836aa2eee2119b0a9a23795883dc99e43/checks/50078765985/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
##[endgroup]
[TIMING:end] tool::ToolBuild { build_compiler: Compiler { stage: 0, host: aarch64-unknown-linux-gnu, forced_compiler: false }, target: aarch64-unknown-linux-gnu, tool: "linkchecker", path: "src/tools/linkchecker", mode: ToolBootstrap, source_type: InTree, extra_features: [], allow_features: "", cargo_args: [], artifact_kind: Binary } -- 0.146
[TIMING:end] tool::Linkchecker { compiler: Compiler { stage: 0, host: aarch64-unknown-linux-gnu, forced_compiler: false }, target: aarch64-unknown-linux-gnu } -- 0.000
##[group]Testing stage1 Linkcheck (aarch64-unknown-linux-gnu)
rustc/platform-support/armv7a-none-eabi.html:191: broken link - `rustc/platform-support/aarch64-unknown-none.html`
rustc/print.html:15249: broken link - `rustc/platform-support/aarch64-unknown-none.html`
checked links in: 22.3s
number of HTML files scanned: 46106
number of HTML redirects found: 14530
number of links checked: 2539436
number of links ignored due to external: 120494
number of links ignored due to exceptions: 24
number of intra doc links ignored: 9
errors found: 2
found some broken links
NOTE: if you are adding or renaming a markdown file in a mdBook, don't forget to register the page in SUMMARY.md
Command `/checkout/obj/build/aarch64-unknown-linux-gnu/stage1-tools-bin/linkchecker /checkout/obj/build/aarch64-unknown-linux-gnu/doc` failed with exit code 1
Created at: src/bootstrap/src/core/build_steps/tool.rs:1550:23
Executed at: src/bootstrap/src/core/build_steps/test.rs:171:77

Command has failed. Rerun with -v to see more details.
Bootstrap failed while executing `--stage 2 test --skip tests --skip coverage-map --skip coverage-run --skip library --skip tidyselftest`
Build completed unsuccessfully in 0:32:56
  local time: Wed Sep 10 20:10:50 UTC 2025
  network time: Wed, 10 Sep 2025 20:10:51 GMT
##[error]Process completed with exit code 1.
```

</details>

thejpster committed:
Add missing links in platform-support.md

thejpster committed:
Add aarch64-unknown-none to SUMMARY.md

thejpster committed:
Update aarch64-unknown-none.md to include -softfloat too

thejpster commented on src/doc/rustc/src/platform-support/aarch64-unknown-none.md:
oops, fixed.

thejpster commented on src/doc/rustc/src/platform-support/aarch64-unknown-none.md:
oops, fixed.

workingjubilee reviewed (changes requested):
I may find more nits but the maintainer assent one is blocking, I think.

*[View changes since this review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/rust/146419/565a9ca63e9df4b223fed0da01f15e578acfb538..409a1fbcba4214b0badf458801038f75fba5c707)*

thejpster committed:
Cleanups from review comments.

thejpster committed:
Insert missing word.

Gelbpunkt reviewed (changes requested):
I only looked at the aarch64 document since that's what I'm most familiar with, most of my comments should also apply to the other documents from what I can tell. I'm very happy about more extensive documentation, this is a fairly good first step.

*[View changes since this review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/rust/146419/565a9ca63e9df4b223fed0da01f15e578acfb538..a0d1f5a3fb8b9554cb4ac5d0b8ffc75233a20dda)*

thejpster committed:
Remove bullet points from the target maintainer list for the arm bare-metal targets

thejpster committed:
Re-order and re-work the aarch64-unknown-none platform support page.

Removes a bunch of information that isn't, strictly speaking, target specific.

workingjubilee commented on src/doc/rustc/src/platform-support/armebv7r-none-eabi.md:
```diff
@@ -0,0 +1,54 @@
+
+## Target maintainers
+
+* [@chrisnc](https://github.com/chrisnc)
```
I have been provided with more information on the context it was assented to, that @chrisnc both is fine with the EDWG helping out and does not want to impede them but neither is he interested in definitely becoming part of that team. I am deciding it is trustworthy enough for this to go ahead, as we can always roll back changes if more information comes to light, I just wanted enough information to be able to presume the situation was correct.

chrisnc replied:
I confirm what @thejpster has said (we corresponded on this topic over email a while back). I personally possess and develop on little- and big-endian v7r platforms with Rust and am available to support the community with these targets as-needed. Thanks for checking!

workingjubilee reviewed (approved):
@thejpster This looks good to go. I believe you are making some final small edits per our conversation, please ping me again when they're in and I'll take a quick glance at them.

*[View changes since this review](https://triagebot.infra.rust-lang.org/gh-changes-since/rust-lang/rust/146419/565a9ca63e9df4b223fed0da01f15e578acfb538..ea661e9353e5dfa2c37f45ad44e52d97d5006447)*

thejpster committed:
Note that these targets are bare-metal.

This is important to note, as it affects how easy it is to build a
binary, and that `#![no_std]` is mandatory.

A different PR should probably add this to all the other platform pages.

workingjubilee commented:
looks good to me!

@bors r+ rollup
